File: src/utils/jwt.ts
import jwt from "jsonwebtoken";
import { JwtPayload } from "jsonwebtoken";
import dotenv from "dotenv";
dotenv.config();

export interface CustomJwtPayload extends JwtPayload {
  userId: string;
  email: string;
}

export const generateToken = (payload: CustomJwtPayload): string => {
  return jwt.sign(payload, process.env.JWT_SECRET as string, {
    expiresIn: "7d",
  });
};

export const verifyToken = (token: string): CustomJwtPayload => {
  try {
    return jwt.verify(
      token,
      process.env.JWT_SECRET as string
    ) as CustomJwtPayload;
  } catch (error) {
    throw new Error("Invalid token");
  }
};

File: python_version/app/utils/jwt.py
from jose import jwt
from datetime import datetime, timedelta
from app.config.settings import settings

ALGORITHM = "HS256"
EXPIRES_DAYS = 7


def generate_token(payload: dict) -> str:
    to_encode = payload.copy()
    expire = datetime.utcnow() + timedelta(days=EXPIRES_DAYS)
    to_encode["exp"] = expire
    return jwt.encode(to_encode, settings.jwt_secret, algorithm=ALGORITHM)


def verify_token(token: str) -> dict:
    try:
        payload = jwt.decode(token, settings.jwt_secret, algorithms=[ALGORITHM])
        return payload
    except Exception:
        raise ValueError("Invalid token")
